src/rdb.c

```diff
@@ -2077,7 +2077,7 @@ robj *rdbLoadObject(int rdbtype, rio *rdb, sds key, int dbid, int *error) {
 
         /* Too many entries? Use a hash table right from the start. */
         if (len > server.hash_max_listpack_entries)
-            hashTypeConvert(o, OBJ_ENCODING_HASHTABLE);
+            hashTypeEnsureHashtableEncoded(o);
```

Comment:
(This and many similar lines.)

I don't think we should do this refactoring for two reasons:

1. We want to avoid unnecessary changes. More touched lines means unnecessary risk of merge conflicts and harder to follow changes using git history and git blame.
2. Just because we currently only convert from listpack to hashtable, it doesn't mean we never will do any other conversion. We might want to convert back to listpack if the hashtable is small and we detect it during some iteration over it.